### ISSUE TITLE
[consolidated] working IMAP strategy

### ISSUE THREAD

**Issue by testbird** [deltachat/deltachat-core#117]
There were several chat <-> email messaging disturbances. For example, regular email traffic is disturbed by chat messages and receipts, if the chat client is not on-line or the chat folder can not be created.

A refined IMAP strategy is explained in the wiki:
https://github.com/deltachat/deltachat-core/wiki/IMAP-strategy

**Implementing this should also solve at least the following issues:**

* Non-INBOX-folders may be synchronized faster https://github.com/deltachat/deltachat-core/issues/12
* Sent messages do not show up (nowhere) in email account https://github.com/deltachat/deltachat-core/issues/15
* Strato Mailbox Integration deltachat/deltachat-android#28
* New Name for the "Chats" folder #47
* delete from server deltachat/deltachat-android#80
* Domain Factory mailbox integration deltachat/deltachat-android#103
* Chats folder not created or used deltachat/deltachat-android#108
* Have the client pull messages after an off-line period deltachat/deltachat-core#84

**And partly**
* Multiple devices on same email address deltachat/deltachat-android#159
* no import of current Chats and Groups https://github.com/deltachat/deltachat-android/issues/210
* Improve multi-device support #52

**Comment by sketet** [comment]
I tried to follow the wiki, but nothing, maybe I don't understand what to do.
I didn't see the old emails in DC, I didn't see the new chat emails...
If DC's SIEVE filter is active? where is the this setting?

**Comment by testbird** [comment]
The ideas in the wiki are to programming improvements, and solving existing issues  (not something that users do), and in the large part not implemented.

If you have or found an issue with deltachat, you are welcome to open a new issue.

**Comment by t-oster** [comment]
Is this the place for discussion on the strategy?
Here are my 5 cents:
- We should only monitor the INBOX and the Chats folder! I would not want DeltaChat to download all my 10 years of email history I sorted into my hirachry of folders. New messages only come to inbox or if correctly recognized into the chat folder. If someone has filters which move mails to different folders, than he has a reason behind this and probably does not want DeltaChat to get those messages anyway. Periodically scanning all folders is a waste of resources in my opinion.
- I am not sure if the SIEVE filters work on every provider, so the fallback is necessary. Also we should test if adding our filter modifies any existing filters (I've seen some docs that only one filter-file can be active at a time, if that is true we possibly mess with the user's created filters)
- Even if the SIEVE filter is active, we need to check the INBOX (Push) because people might respond on chat messages with their Mail client, changing the subject (the filter won't work) but leaving the in-reply-to header intact (DC can recognize the mail, but the filter can't)
- I think the rules under polling/scanning are way to complicated. We should concentrate on the usecase most people have, as stated above: People with advanced filter configurations usually know what they are doing and should adapt their filters if they move DC messages in other folders than INBOX or Chats. IMHO we should not poll at all. This would also remove the need of ignored folders. If the only use-case is to collect Email Adresses, in most cases the addresses are already in the user's address book or in a mail in the inbox.

**Comment by STPKITT** [comment]
What about monitoring the Spam/Junk-folder?

**Comment by t-oster** [comment]
ok the spam/junk also makes sense. but not all my custom and archive folders

**Comment by testbird** [comment]
> if adding our filter modifies any existing filters (I've seen some docs that only one filter-file can be active at a time, if that is true we possibly mess with the user's created filters)

Ok, if only one active filter-file can exist, this removes the problem of file naming and ordering. Then it's just about inserting and maintaining a recongnizable block of lines at the beginning of the active file (and only creating a generically called file if none exists).

> Even if the SIEVE filter is active, we need to check the INBOX

I see, you're right. Ok, you might be able to simplify the push/poll rules with this. So SIEVE can not provide all of the sorting features when deltachat is offline, but it still helps quite a lot for consistent chat separation.

> I think the rules under polling/scanning are way to complicated

They originated from the current behavior, but tried to limit polling (fallback) to only-when-necessary, to ensure that DC does not break for a good part of the users. The problem is that especially the simple (e.g. click/menu create-able, sender dependent filter rules) also catch away chat messages. But if DC always checks if its filter lines are first in the active file, and re-sorts notifies the user about the conflict, you may be able to simplify this.

**Comment by testbird** [comment]
AFAIU collection of addresses (tat you sent to, or replied to) is important to make chat prompts work without an address book requirement. (And polling only necessary with foreign SIEVE filter lines.)

**Comment by t-oster** [comment]
I am unsure about some things here:

- How many providers do allow programatically uploading of SIEVE filters?
- How many average users make use of server-side filters to move unread mails from the inbox? Aren't those already power users, who could change their filters for DeltaChat if needed?
- Can there be more than one SIEVE filter file active at one time?

I myself wasn't aware that a generic concept of server-side mailfilters even exists. I know most people use filters in thunderbird which are client-side IMHO. I also could not find a library in C which allows to upload SIEVE filters, do you have any recommendations on how to get started on that topic?

**Comment by t-oster** [comment]
**domainFactory does not support SIEVE filters** (https://www.df.eu/forum/threads/44113-Imap-SIEVE/page3)

However you can create custom filters in the web-interface, so I can workaround the double-notification-problem (the only reason for sieve filters?) with a custom rule. Thus DC should watch/poll the INBOX,DeltaChat and Junk folders in any case. I would like to have polling of all other folders at least configurable, because I do not have any other filters in place and have > 100 folders.

**Comment by t-oster** [comment]
... do any of the common mail providers support SIEVE? Or just self-hosted dovecot or cyrus-imapd instances (http://www.linux-community.de/blog/wir-wollen-alle-sieve/ german from 2009)

**Comment by testbird** [comment]
Unfortunately, I don't know if common mail providers' make this IMAP feature available to MUAs like firebird with the Sieve plugin, only that server side filtering is the only way that can filter before other clients' notifications. I have used roundcube for this. A well explaining  page seems https://wiki.hostsharing.net/index.php?title=Roundcube_-_Filter

In the absence of imap-sieve support, we should still have a strategy and instructions what the user needs to set up. Something like to always set up push/pull for the inbox AND the chat folder, and instruct users to "set filters at their provider for headers `Chat-Version` and `References: containing mr.thread` into the chat folder"???

http://sieve.info/servers and wikipedia suggested SIEVE is a standard used in quite a number of projects and organizations. Likely more at those looking into, using and promoting a (mobile) instant messenger alternative (DC).

**Comment by t-oster** [comment]
Of cause. I just don't know if it is worth the effort to add the code for automatically creating and uploading the sieve filter if it only works on a small fraction of clients.
However I would love a wiki page with instructions to set up the correct filter in all major web interfaces of common providers.

Also maybe #180 solves parts of the issue

**Comment by testbird** [comment]
To be discussed:
1) What is the correct filter list (criteria+actions)?
1) List of things it solves?

Provider details may come later and with users' experiences.

**Comment by testbird** [comment]
Also to be considered for the filter: An exeption header like `Chat-Version: false`, for example, to be able to move long email replies that evolved from a chat back to the inbox (until further processed by other MUA), and at the same time preventing other DC clients to move the message into the chat folder again.

**Comment by testbird** [comment]
Ok, it's not possible to determine active server filter rules with all providers. But the rules that allow the most flawless operation should be stated, ideally even in the Email-chat spec:

So devs, please review the distillation at https://github.com/deltachat/deltachat-core/wiki/IMAP-strategy#message-handling, taken from above and https://github.com/deltachat/deltachat-core/issues/180?

**Comment by testbird** [comment]
If you say it should work, there only one of the points need higher implementation priority:

- [ ] consider all chat messages not moved out of INBOX as unread/not shown yet 

This should already allow users to set up the server filters to prevent MUA notifications for chat messaging, while having notifications in chat clients.

**Comment by t-oster** [comment]
I am sorry to be the PITA here, but I have some comments:

We have to solve the following problems:
- DC messages should not cause a notification in other MUAs like K9, Thunderbird etc
- DC messages should cause a notification as soon as possible in DC (push/pull/idle)
- Multiple DC instances on the same account should work nicely together
- DC and other MUAs on the same account should work nicely together

It will all boil down to the question: Is it possible to reliably watch/pull/push two IMAP folders at the same time or not.
If this is not possible, the solution from @testbird may be our best shot:
- mark DC messages in the Inbox as read as soon as possible (sieve filter, custom user filter or first DC instance seeing this message)
- Consider seen DC messages in the INBOX not as seen, but only after moved to DC-Folder 

IMHO however this solution feels somewhat unclean, because it does not use the SEEN flag as it is intended and also hinders proper DC+other MUA interop. 

So if it would be possible to watch two folders with the same priority, IMHO the much cleaner way would be:
- All messages identified as DC message are moved to the DC-folder as soon as possible (either by sieve filter, custom filter created by the user or the first DC instance which sees the message). This would fix the issue that other MUAs notify on that message, because in the filter cases the message never arrives in the INBOX
- Use the SEEN flag as it is intended: If you open the message in ANY DC instance, mark it as read, so it appears read in all DC instances and other MUAs. If I am in Thunderbird I can see the DC folder has an unread message, so I could also read it with thunderbird and I'd expect it to be marked as read in DC also
- Watch the Inbox AND DC folder. If the user did not add a filter, DC moves the message as soon as it notices it. If the user add a filter, DC recognizes the message after it is in DC-folder already. DC MSGs which cannot be detected by any filter will still be moved to DC-folder as soon as possible.

This way we would have the following semantics:
- Message is in the DC Folder means the message is identified as DC message
- Messages which are marked SEEN are read by the user with any MUA (including DC)
- Messages marked as UNSEEN are not yet read by the user. If they are in the INBOX they are E-Mails, if they are in the DC-Folder, they are Instant-Messages.

**Comment by testbird** [comment]
Using the fully compliant, reliable solution, if multi-folder watching is possible, would of course be the best.

I'll try to consolidate the wiki with what we have gathered, and will ping for review.

**Comment by t-oster** [comment]
Thanks and again, no offence.

**Comment by testbird** [comment]
No, no, no, it's much appreciated. I explicitly created the wiki document to gather ideas and stipulate discussion. It's much appreciated when others go through it and together we can better improve things.

**Comment by testbird** [comment]
OK, you don´t seem to like the chat client to go through your emails on the server, but the concept of known addresses is rather important in making DC properly work with the least hassle.

Isn't it so that DC doesn't need to download any message bodies, and thanks to the last ID mechanism no data is downloaded if the folder hasn't changed since the last check?

No question, we need a way to have folders ignored that contain addresses that should not get the "known for chatting" status https://github.com/deltachat/deltachat-android/issues/239, maybe that requrement could just be augmented to also support something like "*" (ignore all non-chat and non-spam folders)?

**Comment by t-oster** [comment]
Well I personally do not use the suggested addresses so much. Either I already have them in my contacts or I am answering to a mail in Inbox.
But as I understand it only captures mails which arrive in other folders AFTER the installation. Meaning all my old contacts are not recognized anyway and new mails do usually not arrive in other folder than inbox.

What about a function to refresh the address book which really goes to all messages (without bodies) and collects email adresses which can be started on demand?

This just looks like a waste of resources to me, because everytime an IMAP select etc is issued in folders where no message will ever arrive except i move it the with anoter MUA.
```
10:57:46 0 mails read from "INBOX.Archives.2014.2014-06".
10:57:46 0 mails read from "INBOX.Archives.2014.2014-07".
10:57:47 0 mails read from "INBOX.Archives.2014.2014-12".
10:57:47 0 mails read from "INBOX.Archives.2015".
10:57:47 0 mails read from "INBOX.Archives.2015.2015-01".
10:57:47 0 mails read from "INBOX.Archives.2015.2015-02".
10:57:47 0 mails read from "INBOX.Archives.2015.2015-03".
10:57:47 0 mails read from "INBOX.Archives.2015.2015-04".
10:57:47 0 mails read from "INBOX.Archives.2015.2015-05".
10:57:47 0 mails read from "INBOX.Archives.2015.2015-06".
10:57:47 0 mails read from "INBOX.Archives.2015.2015-07".
10:57:47 0 mails read from "INBOX.Archives.2015.2015-08".
10:57:47 0 mails read from "INBOX.Archives.2015.2015-09".
10:57:47 0 mails read from "INBOX.Archives.2015.2015-10".
10:57:48 0 mails read from "INBOX.Archives.2015.2015-11".
10:57:48 0 mails read from "INBOX.Archives.2015.2015-12".
10:57:48 0 mails read from "INBOX.Archives.2016".
10:57:48 0 mails read from "INBOX.Archives.2016.2016-01".
10:57:48 0 mails read from "INBOX.Archives.2016.2016-02".
10:57:48 0 mails read from "INBOX.Archives.2016.2016-03".
10:57:48 0 mails read from "INBOX.Archives.2016.2016-06".
10:57:48 0 mails read from "INBOX.Archives.2016.2016-07".
10:57:48 0 mails read from "INBOX.Archives.2016.2016-09".
10:57:48 0 mails read from "INBOX.Archives.2016.2016-11".
10:57:48 0 mails read from "INBOX.Archives.2016.2016-12".
10:57:49 Folder "INBOX.Archives.2017" is empty.
10:57:49 0 mails read from "INBOX.Archives.2017".
10:57:49 0 mails read from "INBOX.Archives.2017.2017-01".
10:57:49 0 mails read from "INBOX.Archives.2017.2017-02".
10:57:49 0 mails read from "INBOX.Archives.2017.2017-03".
10:57:49 0 mails read from "INBOX.Archives.2017.2017-04".
10:57:49 0 mails read from "INBOX.Archives.2017.2017-05".
10:57:49 0 mails read from "INBOX.Archives.2017.2017-06".
10:57:49 0 mails read from "INBOX.Archives.2017.2017-07".
10:57:49 0 mails read from "INBOX.Archives.2017.2017-08".
10:57:49 0 mails read from "INBOX.Archives.2017.2017-10".
10:57:49 0 mails read from "INBOX.Archives.2017.2017-11".
10:57:49 0 mails read from "INBOX.Archives.2017.2017-12".
10:57:50 Folder "INBOX.Archives.2018" is empty.
10:57:50 0 mails read from "INBOX.Archives.2018".
10:57:50 0 mails read from "INBOX.Archives.2018.2018-03".
10:57:50 0 mails read from "INBOX.Archives.2018.2018-04".
10:57:50 0 mails read from "INBOX.Auftragsprojekte".
10:57:50 0 mails read from "INBOX.Auftragsprojekte.CircleZing".
10:57:50 0 mails read from "INBOX.Auftragsprojekte.Digitalmedia-OpenImmo".
10:57:50 0 mails read from "INBOX.Auftragsprojekte.Fabiji".
10:57:50 0 mails read from "INBOX.Auftragsprojekte.G-Code".
10:57:50 0 mails read from "INBOX.Auftragsprojekte.Heizung".
10:57:50 0 mails read from "INBOX.Auftragsprojekte.PhillipPraxisAnwesenheit".
10:57:50 0 mails read from "INBOX.Auftragsprojekte.PhotonicCamp".
10:57:51 0 mails read from "INBOX.Auftragsprojekte.PowerNursing".
10:57:51 0 mails read from "INBOX.Auftragsprojekte.PowerNursing.ITSG".
10:57:51 0 mails read from "INBOX.Auftragsprojekte.monticon".
10:57:51 0 mails read from "INBOX.Auftragsprojekte.OsTicket".
10:57:51 0 mails read from "INBOX.Auftragsprojekte.Dienstplan".
10:57:51 0 mails read from "INBOX.Auftragsprojekte.Ingo".
10:57:51 Ignoring "INBOX.Drafts".
10:57:51 0 mails read from "INBOX.Ela".
10:57:51 Folder "INBOX.Entw&APw-rfe" is empty.
10:57:51 0 mails read from "INBOX.Entw&APw-rfe".
10:57:51 0 mails read from "INBOX.Gesendet".
10:57:51 0 mails read from "INBOX.Gleitschirm".
10:57:51 0 mails read from "INBOX.Hochzeit".
10:57:51 0 mails read from "INBOX.Hochzeit.Telekom".
10:57:52 0 mails read from "INBOX.Hochzeit.Treppe".
10:57:52 0 mails read from "INBOX.Hochzeit.Strom".
10:57:52 0 mails read from "INBOX.Hochzeit.E-Bike".
10:57:52 0 mails read from "INBOX.Hochzeit.&ANY-l".
10:57:52 0 mails read from "INBOX.Hochzeit.Baby".
10:57:52 0 mails read from "INBOX.Jobs".
10:57:52 0 mails read from "INBOX.Jobs.AmericanFood".
10:57:52 0 mails read from "INBOX.Jobs.Bewerbungen".
10:57:52 0 mails read from "INBOX.Jobs.FeG".
10:57:52 0 mails read from "INBOX.Jobs.FeG.HomerunNochNichtAbgerechnet".
10:57:52 0 mails read from "INBOX.Jobs.IKA".
10:57:52 0 mails read from "INBOX.Jobs.Jakob-Moreno".
10:57:53 0 mails read from "INBOX.Jobs.Jobangebote".
10:57:53 0 mails read from "INBOX.Jobs.Musik".
10:57:53 0 mails read from "INBOX.Jobs.Musik.Katrenreservierung2018".
10:57:53 0 mails read from "INBOX.Jobs.Nachhilfe".
10:57:53 0 mails read from "INBOX.Jobs.One-Bath".
10:57:53 0 mails read from "INBOX.Jobs.Repaircafe".
10:57:53 0 mails read from "INBOX.Jobs.i10".
10:57:53 Folder "INBOX.Junk" is empty.
10:57:53 0 mails read from "INBOX.Junk".
10:57:53 0 mails read from "INBOX.Kaufen-Verkaufen".
10:57:53 0 mails read from "INBOX.Kaufen-Verkaufen.Amazon".
10:57:53 0 mails read from "INBOX.Kaufen-Verkaufen.OfficeMarion".
10:57:53 0 mails read from "INBOX.Kaufen-Verkaufen.Reichelt".
10:57:54 0 mails read from "INBOX.Kaufen-Verkaufen.eBayKleinanzeigen".
10:57:54 0 mails read from "INBOX.Kaufen-Verkaufen.ebay".
10:57:54 0 mails read from "INBOX.Kaufen-Verkaufen.ebay.nc10".
10:57:54 0 mails read from "INBOX.Kaufen-Verkaufen.nc10".
10:57:54 0 mails read from "INBOX.Kaufen-Verkaufen.C-Saxophon".
10:57:54 0 mails read from "INBOX.Kaufen-Verkaufen.Pitstop".
10:57:54 0 mails read from "INBOX.Lasercutter".
10:57:54 0 mails read from "INBOX.Lasercutter.CircleZING".
10:57:54 0 mails read from "INBOX.Lasercutter.Feedback".
10:57:54 0 mails read from "INBOX.Lasercutter.LAOS".
10:57:54 0 mails read from "INBOX.Lasercutter.Lasersaur".
10:57:55 0 mails read from "INBOX.Lasercutter.iModela".
10:57:55 0 mails read from "INBOX.Lasercutter.Versicherung".
10:57:55 0 mails read from "INBOX.Lexi".
10:57:55 0 mails read from "INBOX.Online Daten".
10:57:55 0 mails read from "INBOX.Online Daten.Auto".
10:57:55 0 mails read from "INBOX.Online Daten.Banking".
10:57:55 0 mails read from "INBOX.Online Daten.Mitgiedschaften".
10:57:55 0 mails read from "INBOX.Online Daten.SSL-Zertifiaket".
10:57:55 0 mails read from "INBOX.Online Daten.Zugangslinks".
10:57:55 0 mails read from "INBOX.Online Daten.Bitcoin".
10:57:55 0 mails read from "INBOX.Papierkorb".
10:57:55 0 mails read from "INBOX.Sent".
10:57:55 Ignoring "INBOX.Spam".
10:57:56 0 mails read from "INBOX.Templates".
10:57:56 Ignoring "INBOX.Trash".
10:57:56 0 mails read from "INBOX.UNI-Sachen".
10:57:56 0 mails read from "INBOX.UNI-Sachen.NachmeldunWebtech".
10:57:56 0 mails read from "INBOX.UNI-Sachen.SS09".
10:57:56 0 mails read from "INBOX.UNI-Sachen.SS09.Proseminar".
10:57:56 0 mails read from "INBOX.UNI-Sachen.SS10".
10:57:56 0 mails read from "INBOX.UNI-Sachen.SS10.JavaChat".
10:57:56 0 mails read from "INBOX.UNI-Sachen.SS10.NXT-Praktikum".
10:57:56 0 mails read from "INBOX.UNI-Sachen.SS11".
10:57:56 0 mails read from "INBOX.UNI-Sachen.SS12".
10:57:56 0 mails read from "INBOX.UNI-Sachen.SS12.GSE".
10:57:57 0 mails read from "INBOX.UNI-Sachen.SS12.PhotonicCamp".
10:57:57 0 mails read from "INBOX.UNI-Sachen.SS13".
10:57:57 0 mails read from "INBOX.UNI-Sachen.WS08".
10:57:57 0 mails read from "INBOX.UNI-Sachen.WS08.AfI-Tutorium".
10:57:57 0 mails read from "INBOX.UNI-Sachen.WS08.Progra".
10:57:57 0 mails read from "INBOX.UNI-Sachen.WS09".
10:57:57 0 mails read from "INBOX.UNI-Sachen.WS10".
10:57:57 0 mails read from "INBOX.UNI-Sachen.WS11".
10:57:57 0 mails read from "INBOX.UNI-Sachen.WS13".
10:57:57 0 mails read from "INBOX.UNI-Sachen.Wichtig".
10:57:57 0 mails read from "INBOX.Urlaub".
10:57:57 0 mails read from "INBOX.Webseiten".
10:57:58 0 mails read from "INBOX.Webseiten.Fartchannel".
10:57:58 0 mails read from "INBOX.Webseiten.Marion".
10:57:58 0 mails read from "INBOX.Webseiten.MusikzugOdenspiel".
10:57:58 0 mails read from "INBOX.Webseiten.V-Server".
10:57:58 0 mails read from "INBOX.Bastelprojekte".
10:57:58 0 mails read from "INBOX.DeltaChat".

**Comment by t-oster** [comment]
Otherwise stated: Wouln't be the only suggested contacts which would not be catched anymore when we only look at Inbox and DeltaChat those, where I have a filter installed to move it away from the inbox? Do you think many people have filters which do move stuff out of the inbox for not known contacts? The contacts I would have special filters for would most certanly be a few special people, which are probably already in my addressbook.

I just tested it: No Mail address of any of my old buddies from the Archive folders appear in DC. So apart from that discussion, maybe we should collect all mailheaders on the first run after DC is set up to build up the address book if that feature is that important.

**Comment by testbird** [comment]
Right, now I remember, part of the conditionals in the the wiki were exactly meant only to poll when server filters deliver emails directly into folders other than INBOX.

> Do you think many people have filters which do move stuff out of the inbox for not known contacts?

I don't think we can know, and applications shouldn't make such assumptions. May there be some trick to determine this, if SIEVE is not available, like a total message count? An example could be mailinglists, or group mails. Any non `FROM` based filter, e.g. `TO`an alias or "+addition".

> it only captures mails which arrive in other folders AFTER the installation.

Yes, I consider this one of the flaws covered in the wiki strategy (scan on init).





(PS: Don't you better want to remove the personal things above?)

**Comment by t-oster** [comment]
I don't think it's feasible to detect the filters automatically (and sieve is not supported that much). 
I think an initial scan is ok or even better a scan which can be started by the user (can take a lot of time and bandwith any may not be desired by every user on first start)

If users have filters active, which do move mails those users want in DC to different folders, I think those users are power-users, who would then adapt their filters. In 99% of the cases I guess those messages are not intended to be recognized in DC at all, but this is of cause my personal feeling.

**Comment by testbird** [comment]
Well I would like the option. Maybe ask: Should delta chat update the list of contacts you interacted with from the emails on the server? (Details Dropdown with list of folders to ignore) And the options: no; only once (now); on every re-connect.

> who would then adapt their filters

How would that work?

**Comment by testbird** [comment]
The same question/modal could be available through the options for manual updates or changes

**Comment by testbird** [comment]
Actually, a selection like the above might actually already serve to select and auto configure the use-case of the user?
https://github.com/deltachat/deltachat-core/wiki/Use-Cases

**Comment by testbird** [comment]
>>   who would then adapt their filters

> How would that work? 

What I mean is, what is a possible way you expect the user adapt their filters?

For example, one may want a project folder to store longer group emails and attachments, and want to be able to chat about them. Deltachat needs to look up the contacts in the group receiving the emails, to allow them poping up in the chat app.

I don't see this as a filter problem, rather than the task of the chat app to support this. (And that's what I think deltachat's contact scanning is good for.)

PS: I updated the explanations for a config selection.

**Comment by testbird** [comment]
Re-reading the options: "Scan only once" actually looks like an expert option that might be left out. (Achievable by first enabling and then disabling the scanning.)

**Comment by testbird** [comment]
Ok, https://github.com/deltachat/deltachat-core/wiki/IMAP-strategy is completed, up to date with the recent disussion.

Please review and improve.
Hopefully the IMAP things is already in your focus anyway due improving the threading.

There are two items marked with **ToDo first** (to quickly allow a separate-notify solution with server filters), I would wish they could be implemented sortly, to have a first version that allows for a configuration that can be used without disturbing MUAs.

Concerning the remaning question in this discussion (filters and scanning for approved receivers), I kept a daily polling / check when foregrounding in the document.

**Comment by testbird** [comment]
Made some updates to [polling and scanning](https://github.com/deltachat/deltachat-core/wiki/IMAP-strategy#polling-and-scanning-for-new-email-approved-contacts).

What do you think about the idea to do the small changes to use the alternative marking and moving first (shortly) to have a working solution even before multi-folder-watch is possible? Or is the multi--watch already getting included with the new threading?

**Comment by testbird** [comment]
Did nobody have the chance to review the wiki? Any comments?

I'd rather not open a separate issue for each point. They are related to many issues, e.g. the missing initial "MsgSCAN" to the latest https://github.com/deltachat/deltachat-android/issues/338.

**Comment by r10s** [comment]
just a short comment, to SIEVE: i do not see that Delta Chat will support SIEVE in the foreseeable future.
to the wiki as such: i think the rules are too complicated - and maybe also too far away from that what is currently _working_ in Delta Chat.

**Comment by testbird** [comment]
I think the current implementation may already cover some parts of the overall strategy (checkmarks), others of course not and some not in the foreseeable future. But it should provide orientation about possibilities and limitations.

Just add comments or questions (as subpoints) in the wiki, that usually helps to flesh out flaws and reasons and understanding.

**Comment by r10s** [comment]
i think some parts are covered by https://github.com/deltachat/deltachat-core/milestone/2
(additional details and information on the mailing list)

**Comment by r10s** [comment]
closing this for now here, of course, the discussion can go on in the new support forum at https://support.delta.chat

**Comment by loziniak** [comment]
Hello. Currently deltachat-core has no wiki. Where can I find the document?

**Comment by testbird** [comment]
There are some wiki postings in the forum:
https://support.delta.chat/t/wiki-imap-strategy-to-chat-within-the-email-ecosystem/123
https://support.delta.chat/t/wiki-use-cases-chat-rules-and-configuration-options/122
